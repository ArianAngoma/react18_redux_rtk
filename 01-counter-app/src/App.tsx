import { useState } from 'react'

interface AppProps {
  value: number
}

function App ({ value }: AppProps): JSX.Element {
  const [counter, setCounter] = useState<number>(value)

  const handleIncrement = (): void => setCounter(counter + 1)

  const handleDecrement = (): void => setCounter(counter - 1)

  const handleReset = (): void => setCounter(value)

  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>

      <button onClick={handleDecrement}>-1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleIncrement}>+1</button>
    </>
  )
}

export default App
