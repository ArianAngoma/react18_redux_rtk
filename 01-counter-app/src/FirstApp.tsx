interface FristAppProps {
  title: string
  subTitle?: string
  name?: string
}

export const FirstApp = ({
  title,
  subTitle = 'No hay subtÃ­tulo',
  name = 'Arian Angoma'
}: FristAppProps) => {

  // console.log(props);

  return (
    <>
      <h1>{title}</h1>
      {/* <code>{ JSON.stringify( newMessage ) }</code> */}
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}
