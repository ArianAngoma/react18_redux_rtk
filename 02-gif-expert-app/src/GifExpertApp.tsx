import React, { useState } from 'react'

const GifExpertApp = () => {
  const [categories, setCategories] = useState<string[]>(['One Punch'])

  const handleAddCategory = (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
    event.preventDefault()

    setCategories(prev => ['HunterXHunter', ...prev])
  }

  return (
    <>
      <h1>GifApp</h1>

      <button
        onClick={handleAddCategory}
        type="submit"
      >
        Agregar
      </button>

      <ol>

        {
          categories.map((category, index) => (
            <li key={index}>{category}</li>
          ))
        }

      </ol>
    </>
  )
}

export default GifExpertApp
