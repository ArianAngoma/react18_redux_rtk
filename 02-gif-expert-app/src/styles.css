* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: rgb(234, 234, 234);
}

body {
    padding: 60px;
}

input {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgb(97, 32, 158);
    color: black;
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 100%;
}

h2 {
    font-size: 1.5rem;
}

h3 {
    font-size: 3rem;
    margin-bottom: 5px;
}

.card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
}

.card{
    align-content: center;
    border: 1px solid grey;
    border-radius: 6px;
    overflow: hidden;
    margin-bottom: 10px;
    margin-right: 10px;
}
.card p{
    padding: 5px;
    text-align: center;
}

.card img{
    max-height: 170px;
}

.card-grid ol{
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    text-align: center;

}

.contenedor{
    max-width: 1200px;
    margin: 0 auto;
}
