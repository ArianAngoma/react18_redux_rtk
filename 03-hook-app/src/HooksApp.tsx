import { useState } from 'react'

function HooksApp () {
  const [counter, setCounter] = useState(0)

  const increment = () => {
    setCounter(counter + 1)
  }

  return (
    <div>
      <h1>Counter: {counter}</h1>
      <button onClick={increment}>Increment</button>
    </div>
  )
}

export default HooksApp
