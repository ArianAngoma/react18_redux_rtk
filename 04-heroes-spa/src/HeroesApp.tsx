import { FC } from 'react'
import AppRouter from './router/AppRouter'
import { AuthProvider } from './auth'

const HeroesApp: FC = () => {
  return (
    <AuthProvider>
      <AppRouter/>
    </AuthProvider>
  )
}

export default HeroesApp
