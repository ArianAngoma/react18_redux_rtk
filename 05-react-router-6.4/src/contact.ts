import localforage from 'localforage'
import { matchSorter } from 'match-sorter'
// @ts-ignore
import sortBy from 'sort-by'

export async function getContacts (query?: string) {
  await fakeNetwork(`getContacts:${query}`)
  let contacts = await localforage.getItem('contacts')
  if (!contacts) contacts = []
  if (query) {
    // @ts-ignore
    contacts = matchSorter(contacts, query, { keys: ['first', 'last'] })
  }
  // @ts-ignore
  return contacts.sort(sortBy('last', 'createdAt'))
}

export async function createContact () {
  // @ts-ignore
  await fakeNetwork()
  let id = Math.random().toString(36).substring(2, 9)
  let contact = {
    id,
    createdAt: Date.now()
  }
  // @ts-ignore
  let contacts = await getContacts()
  // @ts-ignore
  contacts.unshift(contact)
  await set(contacts)
  return contact
}

export async function getContact (id: string) {
  await fakeNetwork(`contact:${id}`)
  let contacts = await localforage.getItem('contacts')
  // @ts-ignore
  let contact = contacts.find(contact => contact.id === id)
  return contact ?? null
}

export async function updateContact (id: string, updates: any) {
  // @ts-ignore
  await fakeNetwork()
  let contacts = await localforage.getItem('contacts')
  // @ts-ignore
  let contact = contacts.find(contact => contact.id === id)
  if (!contact) { // @ts-ignore
    throw new Error('No contact found for', id)
  }
  Object.assign(contact, updates)
  await set(contacts)
  return contact
}

export async function deleteContact (id: string) {
  let contacts = await localforage.getItem('contacts')
  // @ts-ignore
  let index = contacts.findIndex(contact => contact.id === id)
  if (index > -1) {
    // @ts-ignore
    contacts.splice(index, 1)
    await set(contacts)
    return true
  }
  return false
}

function set (contacts: any) {
  return localforage.setItem('contacts', contacts)
}

// fake a cache so we don't slow down stuff we've already seen
let fakeCache = {}

async function fakeNetwork (key: string) {
  if (!key) {
    fakeCache = {}
  }

  // @ts-ignore
  if (fakeCache[key]) {
    return
  }

  // @ts-ignore
  fakeCache[key] = true
  return new Promise(res => {
    setTimeout(res, Math.random() * 800)
  })
}
