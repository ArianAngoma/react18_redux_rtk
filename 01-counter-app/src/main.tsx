import React from 'react'
import ReactDOM from 'react-dom/client'
import App from './App'
import './index.css'
import { FirstApp } from './FirstApp'

ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
  <React.StrictMode>
    {/* <App value={0}/> */}
    <FirstApp title="Hola, soy Arian"/>
  </React.StrictMode>
)
