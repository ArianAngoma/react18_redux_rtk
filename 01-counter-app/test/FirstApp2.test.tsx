import { render, screen } from '@testing-library/react'
import { FirstApp } from '../src/FirstApp'

describe('Tests in <FirstApp />', () => {

  const title = 'Hello, I am a title'
  const subtitle = 'Hello, I am a subtitle'

  test('Should must match with the snapshot', () => {
    const { container } = render(<FirstApp title={title}/>)

    expect(container).toMatchSnapshot()
  })

  test('Should show the title "Hello, I am a title"', () => {
    render(<FirstApp title={title}/>)
    expect(screen.getAllByText(title)).toBeTruthy()
  })

  test('Should show the title in a h1', () => {
    render(<FirstApp title={title}/>)

    expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title)
  })

  test('Should show the subtitle from props', () => {
    render(<FirstApp title={title} subTitle={subtitle}/>)

    expect(screen.getAllByText(subtitle)).toBeTruthy()
    expect(screen.getAllByText(subtitle).length).toBe(2)
  })

})
