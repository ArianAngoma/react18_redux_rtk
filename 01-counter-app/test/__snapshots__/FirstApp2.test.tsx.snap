// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> Should must match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello, I am a title
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Arian Angoma
  </p>
</div>
`;
